src/components/Modals/FiatRamps/fiatRampProviders/onramper.ts

```diff
@@ -0,0 +1,120 @@
+import { adapters, AssetId } from '@shapeshiftoss/caip'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { uniqBy } from 'lodash'
+import { logger } from 'lib/logger'
+
+import { FiatRampAsset } from '../FiatRampsCommon'
+
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'fiatRampProviders', 'OnRamper'],
+})
+
+// Non-exhaustive required types definition. Full reference: https://github.com/onramper/widget/blob/master/package/src/ApiContext/api/types/gateways.ts
+type OnRamperGatewaysResponse = {
+  gateways: GatewayItem[]
+  icons: TokenIconMap
+}
+
+type TokenIconMap = {
+  [key: string]: IconGatewaysResponse
+}
+
+type Currency = {
+  code: string
+  id: string
+  network?: string
+  displayName?: string
+}
+
+type IconGatewaysResponse = {
+  name: string
+  icon: string
+  symbol?: string
+}
+
+type GatewayItem = {
+  identifier: string
+  cryptoCurrencies: Currency[]
+}
+
+export const getOnRamperAssets = async (): Promise<FiatRampAsset[]> => {
+  const data = await (async () => {
```

Comment:
Nitpick: I'm not sure there are any benefits in this being an IIAFE vs. living in the body of `getOnRamperAssets`, if all we're trying to achieve is not to have conflicts with the `data` property from `axios.get()`, a renaming of the destructured property will do the trick.

Nothing wrong in leaving it here though, except from a few additional parens

Reply:
This was copy-pasted from other implementations, but I totally agree - I don't like nested functions, the nesting makes it more difficult to see the context and look at functions in isolation, as they should be. Add JS's variable scope issues and it can get ugly pretty fast. I'll extract this